src/main/java/com/accenture/application/utils/ZipUtils.java

```diff
@@ -0,0 +1,185 @@
+package com.accenture.application.utils;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.log4j.Logger;
+
+public class ZipUtils {
+
+    private static Logger logger = Logger.getLogger(ZipUtils.class);
+
+
+    public File zipFile(String zipOutName, File input, String zippedName) throws FileNotFoundException, IOException {
+        File zip = new File(zipOutName);
+        byte[] buf = new byte[1024];
+        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(zip));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *out*. The resource is referenced in statements at the following lines: 28, 31, 34. The resource closure statement is at lines: 36, 38. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by FileInputStream constructor. Either a) close *out* in a try-finally block or b) close the resource by declaring *out* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).